File: Python3/601 - 800/754 Reach a Number.py
# 36ms, 92.28%; 12.9MB, 7.14%
class Solution:
    def reachNumber(self, target: int) -> int:
        target = abs(target)
        n = int((1 + (1 + 8 * target) ** 0.5) / 2) - 1
        acc = n * (n + 1) // 2
        while acc < target or (acc - target) & 1:
            n += 1
            acc += n
        return n

File: 601 - 800/754 Reach a Number.cc
// 4ms, 100.0%; 8.2MB, 100.0%
#include <bits/stdc++.h>
using namespace std;

static int x = []() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);
    return 0;
}();

class Solution {
public:
    int reachNumber(int target) {
        target = abs(target);
        int n = (1 + sqrt(8) * sqrt(target)) / 2;
        int acc = n * (n + 1) / 2;
        while (acc < target || (acc - target) & 1) ++n, acc += n;
        return n;
    }
};
